Quartz/Engine/Include/Quartz/Core/Engine.hpp

```diff
@@ -23,10 +23,14 @@
 
 #pragma once
 
+#include <Quartz/Core/Core.hpp>
 #include <Quartz/Core/Utilities/Logger.hpp>
 #include <Quartz/Core/Graphics/ContextManager.hpp>
+#include <Quartz/Core/Utilities/EnumTools.hpp>
 
+#define SDL_MAIN_HANDLED
```

Comment:
It may be worth defining this globally in the CMake to avoid having to put it infront of every SDL include. It would also save any potential macro redefinition errors should we ever get in that mess.